### DISCUSSION TITLE
How to style components

### DISCUSSION THREAD

**Discussion by abd3lraouf** [varabyte/kobweb#396] [Q&A] [answered]
I'm trying to customize the color of the Switch widget and this is the only way it worked for me,  I tried using custom styles, modifier with set variable (switch vars ) and it didn't work

```kt
@InitSilk
fun initSiteStyles(ctx: InitSilkContext) {
    ctx.stylesheet.registerStyleBase("body") {
        Modifier
            .fontFamily(
                "-apple-system", "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu",
                "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", "sans-serif"
            )
            .fontSize(18.px)
            .lineHeight(1.5)
    }

    // Silk dividers only extend 90% by default; we want full width dividers in our site
    ctx.theme.modifyComponentStyleBase(HorizontalDividerStyle) {
        Modifier.fillMaxWidth()
    }

    light.run {
        SilkWidgetColorGroups.MutableSwitch(ctx.theme.palettes.light).set(
            thumb = this.controls.colorModeSwitch.thumb,
            backgroundOn = controls.colorModeSwitch.track,
            backgroundOff = Colors.LightGray,
        )
    }

    dark.run {
        SilkWidgetColorGroups.MutableSwitch(ctx.theme.palettes.dark).set(
            thumb = this.controls.colorModeSwitch.thumb,
            backgroundOn = controls.colorModeSwitch.track,
            backgroundOff = Colors.LightGray,
        )
    }
}
```

**Comment by bitspittle** [comment, marked as answer]
The switch widget is unfortunately one of the more complex widgets we provide, but honestly, using the `@InitSilk` method like you did is the best way I would have recommended you do what you want to do.

---

BTW, at some point I introduced a cleaner syntax for updating colors, so you can change your code to:

```kotlin
light.switch.set(...)
dark.switch.set(...)
```

I just updated my own code as well: https://github.com/varabyte/kobweb/commit/e8e50eb04bef956f82492232ddd15f9a3bf16850

**Comment by abd3lraouf** [comment]
Thank you for the clarification, 
BTW I there a way to customize the track border color and width? I thought we could extend the styles you used in theming the Widget, meaning, I could pass the composed variant to the widget which can be composed of all styles used in the component. let me illustrate what I mean.

```
            SwitchTrackStyle.toModifier()
                .setVariable(
                    SwitchVars.TrackBackgroundColor,
                    if (checked) colorScheme?.let { if (colorMode.isDark) it._200 else it._700 }
                        ?: switchPalette.backgroundOn else switchPalette.backgroundOff
                )
                .setVariable(SwitchVars.ThumbColor, thumbColor)
                .setVariable(SwitchVars.FocusColor, focusBorderColor)
                .thenIf(!enabled) { DisabledStyle.toModifier() }
```

This is how you defined the style inside the library,

I mean, there can be a way to customize the Style iteself.

**Comment by abd3lraouf** [comment]
Please never mind, I've added the border to the modifier itself
